modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2094,16 +2094,328 @@ def clip(self, lower, upper, **kwargs):
             new_modin_frame = self._modin_frame.map(lambda df: df.clip(**kwargs))
         return self.__constructor__(new_modin_frame)
 
-    def corr(self, method="pearson", min_periods=1):
-        if method == "pearson":
-            numeric_self = self.drop(
-                columns=[
-                    i for i in self.dtypes.index if not is_numeric_dtype(self.dtypes[i])
-                ]
+    def corr(self, method, min_periods: int = 1):
+        if method != "pearson":
+            return super().corr(
+                method=method, min_periods=min_periods, numeric_only=True
+            )
+
+        def map(df: pandas.DataFrame):
+            # In this kernel we compute all the required components to compute
+            # the correlation matrix at the reduce phase, the required components are:
+            #   1. Matrix holding sums of pairwise multiplications between all columns
+            #      defined as M[col1, col2] = sum(col1[i] * col2[i] for i in range(col_len))
+            #   2. Sum for each column (special case if there are NaN values)
+            #   3. Sum of squares for each column (special case if there are NaN values)
+            #   4. Number of values in each column (special case if there are NaN values)
+            #
+            # It's more convenient to use a numpy array here as it appears to perform
+            # much faster in for-loops which this kernel function has plenty of
+            df = df.select_dtypes(include="number")
+            raw_df = df.values.T
+            nan_mask = np.isnan(raw_df)
+            has_nans = nan_mask.sum() != 0
+
+            if has_nans:
```

Comment:
you have a lot of code under this kind of `if has_nans`, maybe split this function into three - general one (this function), and two handling no-NaN and some-NaN cases separately?

Reply:
moved the corr implementation into a separate file where wrapped the methods in a separate class